packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -497,6 +503,17 @@ export class PreferencesController extends BaseController<
       });
     }
   }
+
+  /**
+   * A setter for the user to opt into smart transactions
+   *
+   * @param smartTransactionsOptInStatus - true to opt into smart transactions
+   */
+  setSmartTransactionsOptInStatus(smartTransactionsOptInStatus: boolean) {
```

Comment:
Nit: will this be potentially used by other controllers? in that case, maybe having an equivalent messenger action will be handy

Reply:
No, it's just a toggle from the Settings in the clients (Mobile only right now).